File: 백준/Silver/1912. 연속합/연속합.py
import sys
from unittest import result 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

n = int(input())
numbers = list(map(int, input().split()))

sum_arr = [0]*len(numbers)
sum_arr[0]=numbers[0]
for i in range(1, len(numbers)):
    sum_arr[i] = max(sum_arr[i-1]+numbers[i], numbers[i])
print(max(sum_arr))

File: 백준/Silver/1912. 연속합/연속합.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var n int
	fmt.Fscanf(reader, "%d\n", &n)
	if n > 0 {
		var numbers = make([]int, n)

		var max = -math.MaxInt
		for i := 0; i < n; i++ {
			fmt.Fscanf(reader, "%d ", &numbers[i])
		}
		var sumArr = make([]int, len(numbers))
		for i, _ := range sumArr {
			sumArr[i] = 0
		}
		sumArr[0] = numbers[0]
		for i := 1; i < len(numbers); i++ {
			sumArr[i] = int(math.Max(float64(sumArr[i-1]+numbers[i]), float64(numbers[i])))
		}
		for _, v := range sumArr {
			if v > max {
				max = v
			}
		}
		fmt.Fprintf(writer, "%d", max)
	} else {
		fmt.Println(0)
	}
}
